### PR TITLE
Add retries to result uploading

### PR DESCRIPTION
**Pull request by LoopedBard3**
Put the upload blob and queue send message into a looping try catch that tries to do the action 3 times with 5 seconds between attempts. This is to hopefully help with individual jobs from the pipeline failing due to Server Busy errors (API limit reached). This was a quick change so am open to other suggestions.

### CODE REVIEW

File: scripts/upload.py [hunk 1]
```diff
@@ -30,14 +31,33 @@ def upload(globpath, container, queue, sas_token_env, storage_account_uri):
...
+                except Exception as ex:
+                    getLogger().error('{0}: {1}'.format(type(ex), str(ex)))
+                    time.sleep(5)
+                    continue
+                
+            if uploadedSuccessfully:
+                for i in range(3):
+                    try:
+                        if queue is not None:
+                            queue_client = QueueClient(account_url=storage_account_uri.format('queue'), queue_name=queue, credential=sas_token, message_encode_policy=TextBase64EncodePolicy())
```
Comment (cincuranet) [inline scripts/upload.py:50]: I guess this could be outside the loop. We don't need to retry constructor.

File: scripts/upload.py [hunk 2]
```diff
@@ -30,14 +31,33 @@ def upload(globpath, container, queue, sas_token_env, storage_account_uri):
...
-            with open(infile, "rb") as data:
-                blob_client.upload_blob(data, blob_type="BlockBlob", content_settings=ContentSettings(content_type="application/json"))
-
-            if queue is not None:
-                queue_client = QueueClient(account_url=storage_account_uri.format('queue'), queue_name=queue, credential=sas_token, message_encode_policy=TextBase64EncodePolicy())
-                queue_client.send_message(blob_client.url)
-
-            getLogger().info("upload complete")
+            uploadedSuccessfully = False
+            for i in range(3):
```
Comment (cincuranet) [inline scripts/upload.py:35]: I would probably create a helper "retry" function that would take lambda and have all the looping and catching encapsulated there.

**Review by cincuranet — COMMENTED** [review]
Is the delay=5s and count=3 based on some measurement or something? The reason I'm asking is because I'm working on similar stuff in _infra_/_perfcommandcore_ and if these values are based on something "real", I would happily ~steal~use same values. :)

File: scripts/upload.py (hunk 1 above)
Comment (LoopedBard3) [reply scripts/upload.py:50]: Nope, These were arbitrarily chosen.

Comment (cincuranet) [reply scripts/upload.py:50]: Fair enough.

File: scripts/upload.py (hunk 2 above)
Comment (LoopedBard3) [reply scripts/upload.py]: Did this, give the new changes a look when you get a chance please. 👍

File: scripts/performance/common.py [hunk 3]
```diff
@@ -157,6 +158,37 @@ def push_dir(path: str = None) -> None:
     else:
         yield
 
+def retry_on_exception(function, retry_count=3, retry_delay=5, retry_delay_multiplier=1, retry_on_exception=None):
```
Comment (cincuranet) [inline scripts/performance/common.py:161]: Would it make sense to have `retry_on_exception=Exception` instead code on L178-179?

File: scripts/upload.py (hunk 2 above)
Comment (cincuranet) [reply scripts/upload.py]: Love the `retry_on_exception`!

File: scripts/performance/common.py (hunk 3 above)
Comment (LoopedBard3) [reply scripts/performance/common.py]: Updated! I agree that this seems like a better approach.

**Review by cincuranet — APPROVED** [review]
Love the `retry_on_exception`. Great work!

**Comment by LoopedBard3** [comment]
Failures are also occurring on main, merging.

**Pull request merged**